kernelci/lava.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/python
+#
+# Copyright (C) 2019 Linaro Limited
+# Author: Dan Rue <dan.rue@linaro.org>
+#
+# This module is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 2.1 of the License, or (at your option)
+# any later version.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+
+def get_device_type_by_name(name, device_types, aliases=[]):
+    """
+        Return a device type named name. In the case of an alias, resolve the
+        alias to a device_type
+
+        Example:
+        IN:
+            name = "x15"
+            device_types = [
+              {'busy': 1, 'idle': 0, 'name': 'am57xx-beagle-x15', 'offline': 0},
+              {'busy': 1, 'idle': 0, 'name': 'beaglebone-black', 'offline': 0},
+              {'busy': 0, 'idle': 1, 'name': 'docker', 'offline': 0},
+              {'busy': 0, 'idle': 1, 'name': 'qemu', 'offline': 0}
+            ]
+            aliases = [
+              {'name': 'x15', 'device_type': 'am57xx-beagle-x15'}
```

Comment:
Are you sure it's not the other way round i.e. `{'name': 'am57xx-beagle-x15', 'device_type': 'x15'}` to match the code in the function?

My point is that if you call the device `x15` in your lab, you need an alias so `am57xx-beagle-x15` also works.  I see the LKFT lab are doing this, using `x15` as the device type with an `am57xx-beagle-x15` alias:
https://lkft.validation.linaro.org/scheduler/device_type/x15

Reply:
My example is backwards, I agree. I can change that so this is less confusing. But I think the code is correct - test case test_get_device_type_by_name_4 hits the code path that resolves an alias to a device_type.

Reply:
OK thanks that makes sense now.